actions/rules/action_state_to_metastate.go

```diff
@@ -234,21 +238,35 @@ func (act ActionStateToMetaState) addOrUpdateChange(changes []convert.Change, at
 	return newChanges
 }
 
+func (act ActionStateToMetaState) fuseChanges(c1 []convert.Change, c2 []convert.Change) *[]convert.Change {
+	for _, change := range c2 {
+		c1 = act.addOrUpdateChange(c1, change.AttributeName, change.OldValue, change.NewValue)
+	}
+	return &c1
+}
+
 // OnChange executes the action rule.
 func (act ActionStateToMetaState) OnChange(newContext convert.ChangeDetector, contextChanges []convert.Change, configuration string, actionChanges *[]convert.Change) (convert.ChangeDetector, []convert.Change, error) {
 	if len(contextChanges) == 0 {
 		// no changes, just return what we have.
 		return newContext, *actionChanges, nil
 	}
+	// if we have multiple changes, this iterates over them and cherrypicks, fusing the results together.
+	var err error
+	var executionChanges []convert.Change
 	for _, change := range contextChanges {
 		if change.AttributeName == workitem.SystemState {
-			return act.OnStateChange(newContext, contextChanges, configuration, actionChanges)
+			newContext, executionChanges, err = act.OnStateChange(newContext, contextChanges, configuration, actionChanges)
 		}
 		if change.AttributeName == workitem.SystemBoardcolumns {
-			return act.OnBoardColumnsChange(newContext, contextChanges, configuration, actionChanges)
+			newContext, executionChanges, err =  act.OnBoardColumnsChange(newContext, contextChanges, configuration, actionChanges)
 		}
+		if err != nil {
+			return nil, nil, err
+		}
+		actionChanges = act.fuseChanges(*actionChanges, executionChanges)
 	}
-	// no changes that match this rule.
+	// return the result
 	return newContext, *actionChanges, nil
```

Comment:
You're not checking for `actionChanges` being `nil` and instead just dereference it.

Reply:
Fixed.